provider/source/src/time_zones/convert.rs

```diff
@@ -685,55 +692,139 @@ impl DataProvider<MetazoneSpecificNamesLongV1> for SourceDataProvider {
             .value
             .dates
             .time_zone_names;
-
         let bcp47_tzid_data = self.iana_to_bcp47_map()?;
         let (meta_zone_id_data, checksum) = self.metazone_to_id_map()?;
         let reverse_metazones = self.reverse_metazones()?;
         let locations = self.calculate_locations(req.id.locale)?.0;
 
         let defaults = iter_mz_defaults(time_zone_names_resource, meta_zone_id_data, true)
-            .flat_map(|(mz, zf)| zone_variant_convert(zf).map(move |(zv, v)| ((mz, zv), v)))
-            .filter(|&((mz, zv), v)| {
-                let Some(tzs) = reverse_metazones.get(&mz) else {
-                    log::warn!("No tzs for {mz:?}");
-                    return false;
-                };
-                tzs.iter().any(|tz| {
+            .filter_map(|(mz, zf)| {
```

Comment:
suggestion: would be cool to share code between this and the other one, if possible

Reply:
I think this is easier to follow